static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -0,0 +1,1033 @@
+import {Fragment, PropsWithChildren, ReactNode, useState} from 'react';
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {ModalRenderProps, openModal} from 'sentry/actionCreators/modal';
+import Alert from 'sentry/components/alert';
+import {CodeSnippet} from 'sentry/components/codeSnippet';
+import {FeedbackModal} from 'sentry/components/featureFeedback/feedbackModal';
+import ExternalLink from 'sentry/components/links/externalLink';
+import Link from 'sentry/components/links/link';
+import ProgressRing from 'sentry/components/progressRing';
+import {TabPanels, Tabs} from 'sentry/components/tabs';
+import {TabList} from 'sentry/components/tabs/tabList';
+import {
+  IconCheckmark,
+  IconCircle,
+  IconMegaphone,
+  IconOpen,
+  IconQuestion,
+  IconRefresh,
+  IconWarning,
+} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+interface SourceResoultionResults {
+  distName: string | null;
+  eventHasDebugIds: boolean;
+  matchingArtifactName: string;
+  projectHasUploadedArtifacts: boolean;
+  releaseName: string | null;
+  releaseSourceMapReference: string | null;
+  sdkDebugIdSupport: 'full' | 'needs-upgrade' | 'unofficial-sdk';
+  sourceFileReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceFileScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  sourceMapReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceMapScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  stackFrameDebugId: string | null;
+  stackFramePath: string | null;
+  uploadedSomeArtifact: boolean;
+  uploadedSomeArtifactToRelease: boolean;
+  uploadedSomeArtifactWithDebugId: boolean;
+  uploadedSourceFileWithCorrectDebugId: boolean;
+  uploadedSourceMapWithCorrectDebugId: boolean;
+  sdkVersion?: string;
+}
+
+interface SourceMapsDebuggerModalProps extends ModalRenderProps {
+  sourceResolutionResults: SourceResoultionResults;
+}
+
+export function SourceMapsDebuggerModal({
+  Body,
+  Header,
+  Footer,
+  sourceResolutionResults,
+}: SourceMapsDebuggerModalProps) {
+  const theme = useTheme();
+
+  let debugIdProgress = 0;
+  if (sourceResolutionResults.sdkDebugIdSupport === 'full') {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.stackFrameDebugId !== null) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceFileWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceMapWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  const debugIdProgressPercent = debugIdProgress / 4;
```

Comment:
l: can we extract into function? I don't think component body needs knowledge of `debugIdProgress`, and makes it easier to understand which `sourceResolutionResults` attributes are needed to determine individual progress.

Reply:
👍